### PR TITLE
Made BehaviorParameters internal

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ enum BehaviorType
         /// </summary>
         [HideInInspector]
         [SerializeField]
-        public int m_TeamID;
+        public int TeamId;
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs]: `[FormerSerializedAs...]`?

File: com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs [hunk 2]
```diff
@@ -57,7 +57,7 @@ enum BehaviorType
         /// <summary>
         /// The associated <see cref="BrainParameters"/> for this behavior.
         /// </summary>
-        internal BrainParameters brainParameters
+        public BrainParameters brainParameters
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs]: Is there any point to making this public if BehaviorParameters and BrainParameters are both internal?

Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs]: All public fields on internal classes are internal

File: com.unity.ml-agents/Runtime/Agent.cs [hunk 3]
```diff
@@ -119,6 +119,13 @@ internal struct AgentParameters
             public int maxStep;
         }
 
+        public int TeamId {
+            get { return GetComponent<BehaviorParameters>().TeamId; }
+        }
+        public string BehaviorName  {
```
Comment (vincentpierre) [inline com.unity.ml-agents/Runtime/Agent.cs]: This is new API, Is that okay ?

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Agent.cs]: Yeah, I think it's fine. Only question is whether we should cache the BehaviorParameters - I believe `GetComponent<>` has non-trivial cost.

File: com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs (hunk 1 above)
Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs]: added

File: com.unity.ml-agents/Runtime/Agent.cs (hunk 3 above)
Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/Agent.cs]: Ok, I now use the cached BehaviorParameters

**Comment by MikeWise2718** [comment]
Making the BehaviorParameter internal, and all its fields internal breaks my code, since I like to build agents programmatically. This means I add the components with script and set the parameters that way too. This feels to me like a more scalable and testable approach. Unless I am overseeing something (which is entirely possible), this means I now have to base everything on a hand constructed prefabs.  

Isn't this a pretty sever limitation?

**Pull request merged**